qiita_pet/templates/artifact_ajax/processing_artifact.html

```diff
@@ -203,67 +213,140 @@
    * @param p_name str the name of the parameter
    * @param param_info object the information of the parameter
    * @param sel_artifacts_info object with the information of the currently selected artifacts
+   * @param target DOM div to add the parameter gui
+   * @param dflt_val object with the default value to use for the given parameter
+   * @param allow_change_optionals bool whether to allow changing the default optional parameters or not
    *
    * This function generates the needed GUI specific to the given parameter type
    *
    */
-  function load_parameter_gui(p_name, param_info, sel_artifacts_info) {
-    $('<i>').appendTo("#cmd-opts-div").text(p_name.replace('_', ' ') + ': ');
+  function load_parameter_gui(p_name, param_info, sel_artifacts_info, target, dflt_val, allow_change_optionals) {
+    // Create the parameter interface
+    var $rowDiv = $('<div>').addClass('row').addClass('form-group').appendTo(target);
+    // Replace the '_' by ' ' in the parameter name for readability
+    $('<label>').addClass('col-sm-2').addClass('col-form-label').text(p_name.replace('_', ' ') + ': ').appendTo($rowDiv).attr('for', p_name);
+    var $colDiv = $('<div>').addClass('col-sm-3').appendTo($rowDiv);
+
     var p_type = param_info[0];
     var allowed_types = param_info[1];
-    if (p_type == 'artifact') {
-      var sel = $('<select>').appendTo("#cmd-opts-div").attr('id', p_name).attr('name', p_name).addClass('required-parameter');
-      for(var key in sel_artifacts_info) {
-         if(allowed_types.indexOf(sel_artifacts_info[key].type) !== -1) {
-           sel.append($('<option>').attr('value', key).text(sel_artifacts_info[key].name));
-         }
+    var $inp;
```

Comment:
I'm not sure either which one is better, just I'm used to do this as a C programmer. If you feel strong about it I can change.